### PR TITLE
Add amun liquidity adapters

### PR DESCRIPTION
**Pull request by loki-sama**
Adds Amun liquidity simple and interactive adapters
Adds tests for adapters

### CODE REVIEW

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        address liquidityToken = abi.decode(data, (address));
+
+        address token = tokenAmounts[0].token;
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+
+        tokensToBeWithdrawn = new address[](1);
+        tokensToBeWithdrawn[0] = liquidityToken;
+
+        ERC20(token).safeApprove(liquidityToken, amount, "LBIA");
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol:62]: please use `safeApproveMax` here

**Comment by sobolev-igor** [comment]
Here everything seems to be correct, except for `safeApproveMax` function call

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol (hunk 1 above)
Comment (loki-sama) [reply contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol]: I updated to safeApproveMax

**Comment by loki-sama** [comment]
There is only one liquidity token: '0x9D895763488a9f60D1882296cBe23399f974E10d'

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import { IVault } from "../../interfaces/IVault.sol";
+
+/**
+ * @title Interactive adapter for AmunLiquidity.
+ * @dev Implementation of InteractiveAdapter abstract contract.
+ * @author Timo <Timo@amun.com>
+ */
+contract AmunLiquidityInteractiveAdapter is InteractiveAdapter, ERC20ProtocolAdapter {
+    using SafeERC20 for ERC20;
+    uint8 referralCode = 101;
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol:35]:
```suggestion
    uint8 internal constant REFERRAL_CODE = 101;
```

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+        address token = tokenAmounts[0].token;
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+
+        tokensToBeWithdrawn = new address[](1);
+        tokensToBeWithdrawn[0] = liquidityToken;
+
+        ERC20(token).safeApproveMax(liquidityToken, amount, "LBIA");
+        // solhint-disable-next-line no-empty-blocks
+        try IVault(liquidityToken).deposit(amount, referralCode)  {} catch Error(
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol:64]:
```suggestion
        try IVault(liquidityToken).deposit(amount, REFERRAL_CODE)  {} catch Error(
```

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        require(tokenAmounts.length == 1, "LBIA: should be 1 tokenAmount[2]");
+
+        address token = tokenAmounts[0].token;
+        uint256 amount = getAbsoluteAmountWithdraw(tokenAmounts[0]);
+
+        tokensToBeWithdrawn = new address[](1);
+        tokensToBeWithdrawn[0] = IVault(token).underlying();
+
+        // solhint-disable-next-line no-empty-blocks
+        try IVault(token).withdraw(amount, referralCode)  {} catch Error(string memory reason) {
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol:95]:
```suggestion
        try IVault(token).withdraw(amount, REFERRAL_CODE)  {} catch Error(string memory reason) {
```

**Review by sobolev-igor — CHANGES_REQUESTED** [review]

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol (hunk 2 above)
Comment (loki-sama) [reply contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol]: 👍

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import { IVault } from "../../interfaces/IVault.sol";
+
+/**
+ * @title Interactive adapter for AmunLiquidity.
+ * @dev Implementation of InteractiveAdapter abstract contract.
+ * @author Timo <Timo@amun.com>
+ */
+contract AmunLiquidityInteractiveAdapter is InteractiveAdapter, ERC20ProtocolAdapter {
+    using SafeERC20 for ERC20;
+    uint8 internal constant REFERRAL_CODE = 101;
```
Comment (sobolev-igor) [inline contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol:35]:
```suggestion
    uint16 internal constant REFERRAL_CODE = 101;
```

**Review by sobolev-igor — CHANGES_REQUESTED** [review]

File: contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol (hunk 5 above)
Comment (loki-sama) [reply contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol]: 👍  Thanks

**Pull request merged**